EIPS/eip-7867.md

```diff
@@ -0,0 +1,553 @@
+  critical call if the batch requests an atomicity level the wallet cannot
+  provide for any reason other than user rejection.
+  * Wallets supporting `strict` but not `loose` SHOULD NOT reject `loose`
+    batches and SHOULD instead upgrade the request to strict atomicity.
+  * Note that a batch with exactly one call _always_ satisfies the requirements
+    of strict atomicity.
+* MUST reject (with error code 0<!-- TODO -->) batches containing unsupported
+  `on-failure` modes.
+* MUST reject (with error code 0<!-- TODO -->) batches containing unsupported
+  combinations/orderings of call-scope `on-failure` modes.
+    * Wallets MUST reject `rollback` when used in a `none` batch, even if the
+      batch is upgraded to `loose` or `strict` atomicity. This also applies to
+      calls that do not specify an explicit `on-failure` mode.
+* MAY reject (with error code 0<!-- TODO -->) the request if the batch is
+  expected to be rolled back.
+* SHOULD inform the user before executing any calls if any call in the batch is
+  expected to fail.
+
+
+#### `wallet_getCallsStatus`
+
+When `wallet_getCallsStatus` is called with a batch identifier corresponding to
+a batch submitted with the batch-scope `flow-control` capability enabled, the
+following changes override the behaviour defined in EIP-5792. Note that:
+
+* There are no changes when called with a batch without this capability enabled;
+  and
+* Even if the behaviour of the batch is not changed from the default (eg.
+  setting `atomicity` to `strict` and omitting the `flow-control` capability for
+  all calls), the following changes still apply.
+
+##### Removed Requirements
+
+These requirements defined in EIP-5792 are removed:
+
+> * If a wallet executes multiple calls **atomically** in a single transaction,
+>   `wallet_getCallsStatus` MUST return an object with a `receipts` field that
+>   contains a single transaction receipt, corresponding to the transaction
+>   in which the calls were included.
+> * If a wallet executes multiple calls **non-atomically** through some
+>   `capability` defined elsewhere, `wallet_getCallsStatus` MUST return an
+>   object with a `receipts` field that contains **an array of receipts** for
+>   all transactions containing batch calls that were included onchain. This
+>   includes the batch calls that were included on-chain but eventually
+>   reverted.
+
+
+##### Added Requirements
+
+###### Capabilities
+
+The returned capabilities object:
+
+* MUST contain a `flow-control` key set to exactly the boolean `true`.
+    * It may be tempting to include additional detail about the status of
+      individual calls here, but don't. Instead use a multi-status capability
+      defined elsewhere.
+
+###### Receipts
+
+The returned `receipts` array:
+
+* MUST NOT contain more than one receipt for the same transaction.
+* SHOULD NOT contain receipts for transactions without a call from the requested
```

Comment:
Why not MUST NOT? Is it allowed to add a receipt for a call that was not icluded in the batch? If yes, why, and what is the use case?

Reply:
I went with "MUST NOT" for things that would break/be unsafe, and SHOULD NOT for things that would be weird but not dangerous.